account-kit/smart-contracts/src/light-account/clients/multiOwnerLightAccount.ts

```diff
@@ -71,14 +100,43 @@ export function createMultiOwnerLightAccountClient<
  * });
  * ```
  *
- * @param {CreateMultiOwnerLightAccountClientParams} params the configuration for creating the multi-owner light account client
+ * @example
+ * ```ts
+ * import { createMultiOwnerLightAccountClient } from "@account-kit/smart-contracts";
+ * import { sepolia, alchemy } from "@account-kit/infra";
+ * import { LocalAccountSigner } from "@aa-sdk/core";
+ * import { generatePrivateKey } from "viem"
+ *
+ * const lightAccountClient = await createMultiOwnerLightAccountClient({
+ *  transport: alchemy({
+ *    apiKey: "your-api-key",
+ *  }),
+ *  chain: sepolia
+ *  signer: LocalAccountSigner.privateKeyToAccountSigner(generatePrivateKey())
+ * });
+ * ```
+ *
+ * @param {CreateMultiOwnerLightAccountClientDynamicTransportParams} params the configuration for creating the multi-owner light / alchemy account client with the provided parameters transport
  * @returns {Promise<SmartAccountClient>} a promise that resolves to a `SmartAccountClient` containing the created account client and relevant methods
  */
 export async function createMultiOwnerLightAccountClient(
-  params: CreateMultiOwnerLightAccountClientParams
-): Promise<SmartAccountClient> {
+  params: CreateMultiOwnerLightAccountClientDynamicTransportParams
+): Promise<SmartAccountClient | AlchemySmartAccountClient> {
   const { transport, chain } = params;
 
+  if (isAlchemyTransport(transport, chain)) {
+    const account = await createMultiOwnerLightAccount({
+      ...params,
+      transport,
+      chain,
+    });
```

Comment:
same comment here about extracting out of if statement

Reply:
Hoisted